pkg/webhook/sidecar_spec.go

```diff
@@ -118,6 +121,70 @@ func GetSidecarContainerSpec(c *Config) corev1.Container {
 	return container
 }
 
+func (si *SidecarInjector) GetNativeMetadataPrefetchSidecarContainerSpec(pod *corev1.Pod, c *Config) corev1.Container {
+	container := si.GetMetadataPrefetchSidecarContainerSpec(pod, c)
+	container.Env = append(container.Env, corev1.EnvVar{Name: "NATIVE_SIDECAR", Value: "TRUE"})
+	container.RestartPolicy = ptr.To(corev1.ContainerRestartPolicyAlways)
+
+	return container
+}
+
+func (si *SidecarInjector) GetMetadataPrefetchSidecarContainerSpec(pod *corev1.Pod, c *Config) corev1.Container {
+	limits, requests := prepareResourceList(c)
+
+	// The sidecar container follows Restricted Pod Security Standard,
+	// see https://kubernetes.io/docs/concepts/security/pod-security-standards/#restricted
+	container := corev1.Container{
+		Name:            SidecarMetadataPrefetchName,
+		Image:           c.MetadataContainerImage,
+		ImagePullPolicy: corev1.PullPolicy(c.ImagePullPolicy),
+		SecurityContext: &corev1.SecurityContext{
+			AllowPrivilegeEscalation: ptr.To(false),
+			ReadOnlyRootFilesystem:   ptr.To(true),
+			Capabilities: &corev1.Capabilities{
+				Drop: []corev1.Capability{
+					corev1.Capability("ALL"),
+				},
+			},
+			SeccompProfile: &corev1.SeccompProfile{Type: corev1.SeccompProfileTypeRuntimeDefault},
+			RunAsNonRoot:   ptr.To(true),
+			RunAsUser:      ptr.To(int64(NobodyUID)),
+			RunAsGroup:     ptr.To(int64(NobodyGID)),
+		},
+		Args: []string{
+			"--v=5",
+		},
+		Resources: corev1.ResourceRequirements{
+			// We should change these resources.
```

Comment:
we need to use specific resources here. im not sure if its worth setting up a pipeline to pass in custom resources here. for v1, id say no. Id probably just hardcode them.

Reply:
we need to test how many resources ls is using

Reply:
we do not have a flow setup on OSS to use configmaps for setting sidecar spec. Hardcoding these in non-managed driver.

Reply:
Sidecar resources should be enough here based on testing and minimum requirements. Resolving...